src/v/storage/fs_utils.cc

```diff
@@ -106,7 +106,7 @@ segment_full_path segment_full_path::to_compacted_index() const {
     if (extension == ".log") {
         return with_extension(".compaction_index");
     } else if (extension == ".log.compaction.staging") {
-        return with_extension("log.compaction.compaction_index");
+        return with_extension(".log.compaction.compaction_index");
```

Comment:
does the affect the ability of storage to clean-up the transient files? i'm not sure what kind of wildcard / pattern it uses to identify them.

Reply:
It would if we cleaned up leftover files at start up, but I don't think we do currently. We only clean them up at runtime after we compact

Reply:
Good to merge? If so, please do. It's a thrill :D